multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/IncrementalAppInstancesUpdateStep.java

```diff
@@ -13,93 +13,137 @@
 import org.cloudfoundry.multiapps.controller.core.model.DeployedMtaApplication;
 import org.cloudfoundry.multiapps.controller.core.model.ImmutableIncrementalAppInstanceUpdateConfiguration;
 import org.cloudfoundry.multiapps.controller.core.security.token.TokenService;
+import org.cloudfoundry.multiapps.controller.process.Messages;
 import org.cloudfoundry.multiapps.controller.process.variables.Variables;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.config.BeanDefinition;
 import org.springframework.context.annotation.Scope;
 
 import com.sap.cloudfoundry.client.facade.CloudControllerClient;
 import com.sap.cloudfoundry.client.facade.domain.CloudApplication;
 import com.sap.cloudfoundry.client.facade.domain.InstanceInfo;
+import com.sap.cloudfoundry.client.facade.domain.InstanceState;
 
 @Named("incrementalAppInstancesUpdateStep")
 @Scope(BeanDefinition.SCOPE_PROTOTYPE)
 public class IncrementalAppInstancesUpdateStep extends TimeoutAsyncFlowableStep {
 
-    @Inject
-    protected CloudControllerClientFactory clientFactory;
-    @Inject
-    protected TokenService tokenService;
+    private static final Logger LOGGER = LoggerFactory.getLogger(IncrementalAppInstancesUpdateStep.class);
 
-    @Override
-    public Duration getTimeout(ProcessContext context) {
-        return context.getVariable(Variables.START_TIMEOUT);
+    private final CloudControllerClientFactory clientFactory;
+    private final TokenService tokenService;
+
+    @Inject
+    public IncrementalAppInstancesUpdateStep(CloudControllerClientFactory clientFactory, TokenService tokenService) {
+        this.clientFactory = clientFactory;
+        this.tokenService = tokenService;
     }
 
     @Override
     protected StepPhase executeAsyncStep(ProcessContext context) throws Exception {
```

Comment:
I'd suggest extracting some of the logic into other methods to make this one more concise.

Reply:
done